functions.php

```diff
@@ -53,7 +61,7 @@ function createRecipeCards(array $recipes): string {
  * @return string
  */
 function linkValidation(string $error): string {
-    if (!filter_var($_POST['link'], FILTER_VALIDATE_URL)) {
+    if (!filter_var($_POST['addLink'], FILTER_VALIDATE_URL)) {
```

Comment:
undefined return, either define error before the if statement or return in the if.

Reply:
I've defined the error outside the function